pkg/service/service.go

```diff
@@ -639,6 +659,33 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 	return new, lb.ID(svc.frontend.ID), nil
 }
 
+// filterServiceBackends returns the list of backends based on given front end ports.
```

Comment:
The `pkg/service` changes LGTM, just I have one concern adding the new code to the package - we have discussed that the L7 related changes should not live in the `pkg/service` package (the first item in https://github.com/cilium/cilium/issues/19698). Do you see the change happening in the nearest future?